airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/BaseRecordBufferImplementation.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.record_buffer;
+
+import com.google.common.io.CountingOutputStream;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URISyntaxException;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class BaseRecordBufferImplementation implements RecordBufferImplementation {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BaseRecordBufferImplementation.class);
+
+  private final RecordBufferStorage bufferStorage;
+  private final CountingOutputStream byteCounter;
+  private final GzipCompressorOutputStream compressedBuffer;
+
+  private InputStream inputStream;
+  private boolean isClosed;
+
+  protected BaseRecordBufferImplementation(final RecordBufferStorage bufferStorage, final boolean withCompression) throws Exception {
```

Comment:
are we going to add a `withEncryption` boolean here to support encryption?

Reply:
I guess yes.

Maybe we should move them out of the constructor and have a dedicated `.withCompression(boolean)` or `.withEncryption(enum algorithm)` method instead?